src/map.rs

```diff
@@ -60,14 +58,25 @@ macro_rules! load_factor {
 
 /// A concurrent hash table.
 ///
+/// Note that `ahash::RandomState`, the default value of `S`, is not
+/// cryptographically secure. Therefore it is strongly recommended that you do
+/// not use this hash for cryptographic purproses.
+/// See [`ahash`](https://github.com/tkaitchuck/ahash) for more information.
+///
 /// See the [crate-level documentation](index.html) for details.
-pub struct HashMap<K, V, S = RandomState> {
+#[cfg(feature = "std")]
+pub struct HashMap<K, V, L = parking_lot::RawMutex, S = RandomState>
```

Comment:
I suggest `L` and `S` should be reversed, so `HashMap<K, V, S>` still looks like the standard collection. This would also simplify the test cases where you had to start specifying the lock type to use a different hasher, which feels like an API wart. However, that means you'd have to remove the default `S` in the no-std configuration too.

Reply:
Yeah, not wanting to remove the default `S` in no-std is why I ordered them this way. I'm honestly not sure which is better/worse. My inclination is to make things harder for `no_std` folks when given a trade-off, which would suggest no defaults in no-std and `L` last.